rust/src/ssh/ssh.rs

```diff
@@ -179,7 +194,7 @@ impl SSHState {
                             let remlen = rem.len() as u32;
                             hdr.record_left = head.pkt_len - 2 - remlen;
                             //header with rem as incomplete data
-                            if head.msg_code == SSH_MSG_NEWKEYS {
+                            if head.msg_code == parser::MessageCode::SshMsgNewKeys {
```

Comment:
Did you try to split the TCP packet with the hassh data in 2 packets ?

I think that this case should be handled here somehow

Reply:
Yes, it looks like this pcap is good for testing.